code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -212,14 +212,24 @@
 	become_hearing_sensitive(trait_source = ROUNDSTART_TRAIT)
 	update_step_speed()
 
-/obj/vehicle/sealed/mecha/Destroy()
+//separate proc so that the ejection mechanism can be easily triggered by other things, such as admins
+/obj/vehicle/sealed/mecha/proc/Eject()
 	for(var/M in occupants)
 		var/mob/living/occupant = M
 		if(isAI(occupant))
 			occupant.gib() //No wreck, no AI to recover
 		else
+			occupant.Stun(2 SECONDS)
+			occupant.Knockdown(destruction_knockdown_duration)
+			occupant.throwing = TRUE //This is somewhat hacky, but is the best option available to avoid chasm detection for the split second between the next two lines
 			occupant.forceMove(loc)
-			occupant.SetSleeping(destruction_sleep_duration)
+			occupant.throw_at(get_edge_target_turf(src,pick(GLOB.alldirs)),rand(5, 8),rand(5, 8)) //resets the throwing variable above.
```

Comment:
You need to have a variable(i.e. `var/eject_value`) to save `rand(5, 8)`, and use that into the two augments because those will get different value but I guess you wanted the same result

Reply:
can you explain why I need to declare a variable when I want random force and random distance? The two values don't have to be the same - this determines how much damage the ejection does (upon collision), and how far it goes independently. 

I don't particularly want these to share one value.

Reply:
> when I want random force and random distance?

that you used the same `rand()` values looked like that you wanted to sync the value of thrown_speed and distance originally.
In this case, you'd need to leave a comment that the different rand values are intended because this looks like a typical mistake of someone when they want the same value but put `rand()` on desired positions.